src/GraphQL/Types/GraphTypesLookup.cs

```diff
@@ -382,6 +384,7 @@ private void HandleField(IComplexGraphType parentType, FieldType field, TypeColl
                 if (applyNameConverter)
                 {
                     arg.Name = NameConverter.NameForArgument(arg.Name, parentType, field);
+                    NameValidator.ValidateName(arg.Name, "argument");
```

Comment:
just move this check into setter

Reply:
That’s impossible. Or at least it entirely defeats the purpose of this PR. The NameConverter has not been applied when the setter runs, and nor has the field or argument been added to a schema.

Reply:
I am talking about `NameValidator.ValidateName` check, not `NameConverter.NameForArgument` or I don't understand. What are we trying to achieve? Guaranteed validation of names? If so it is not needed to call `NameValidator.ValidateName` anywhere else but inside setter.

> The NameConverter has not been applied when the setter runs

?

`arg.Name = NameConverter.NameForArgument(arg.Name, parentType, field);` means apply converter and call setter, does not ?

Reply:
That is not the goal. The goal is to validate after the NameConverter has been applied, **and not before**.

Reply:
If the check is in the setter, it gets checked before and after.

Reply:
>  The goal is to validate after the NameConverter has been applied, and not before.

Not before, let's say, OK. But checking inside setter covers both cases.

Reply:
What's the point in these checks at all, if you can always just take any field and directly change its name to invalid value at any time?

Reply:
We can write another PR for that issue. But most of the codebase relies on the fact that people don’t make changes to the schema after it has been initialized. And this PR covers that scenario, and even moreso than the current codebase.

Reply:
> > The goal is to validate after the NameConverter has been applied, and not before.
> 
> Not before, let's say, OK. But checking inside setter covers both cases.

Checking inside setter covers before AND after. Not ‘after and not before’, which is what we are after here.

Reply:
So it might be worth changing the goal ?

> We can write another PR for that issue.

I don't see the point in doing this because the proposed changes are very small. The new PR will remove some of the edits of this one. Why do we need this?

Reply:
See this https://github.com/graphql-dotnet/graphql-dotnet/pull/1961#discussion_r513394623

Reply:
I don’t mind moving checks into the setters, if it fits with the goal of this PR. I would normally keep separate PRs to fix separate issues. Personally I think moving the checks into the setters has little practical value over the existing design, considering the design of the rest of the codebase (nothing is protected against runtime changes).

Reply:
The current design is far from perfect and will inevitably undergo changes. This is not a reason not to move towards ensuring schema correctness.